src/array.jl

```diff
@@ -970,7 +970,9 @@ Base.repeat(a::CatArrOrSub{T, N};
 StructTypes.StructType(::Type{<:CategoricalVector}) = StructTypes.ArrayType()
 
 function StructTypes.construct(::Type{<:CategoricalArray}, A::Vector)
-    if eltype(A) >: Nothing
+    if eltype(A) == Any
+        categorical(identity.(ifelse.(A .=== nothing, missing, A)))
```

Comment:
`identity` is only needed if there's no function to call, but here you have `ifelse`.
```suggestion
        categorical(ifelse.(A .=== nothing, missing, A))
```